tests/integration/test_using_cached_dependencies.py

```diff
@@ -1,86 +1,117 @@
-    * Send new request to Cachito API which would fetch data from the prepared repository
-    * Delete branch with the corresponding commit
-    * Send the same request to Cachito API
-
-    Checks:
-    * Check that the state of the first request is complete
-    * Check that the commit is not available in the repository after the branch is deleted
-    * Check that the state of the second request is complete
-    """
-    generated_suffix = "".join(
-        random.choice(string.ascii_letters + string.digits) for x in range(10)
-    )
-    branch_name = f"test-{generated_suffix}"
-    repo = git.repo.Repo.clone_from(
-        test_env["cached_dependencies"]["seed_repo"]["https_url"], tmpdir
-    )
-    remote = repo.create_remote("test", url=test_env["cached_dependencies"]["test_repo"]["ssh_url"])
-    assert remote.exists()
-
-    try:
-        repo.create_head(branch_name).checkout()
-        repo.git.commit("--allow-empty", m="Commit created in integration test for Cachito")
-        repo.git.push("-u", remote.name, branch_name)
-        commit = repo.head.commit.hexsha
-
-        client = utils.Client(
-            test_env["api_url"], test_env["api_auth_type"], test_env.get("timeout")
+class TestCachedDependencies:
+    """Test class for cached dependencies."""
+
+    def setup_method(self, method):
+        """Create pool for removing shared directories."""
+        self.directories = []
+
+    def teardown_method(self, method):
+        """Remove shared directories in the pool."""
+        for directory in self.directories:
+            shutil.rmtree(directory)
+
+    def test_using_cached_dependencies(self, test_env, tmpdir):
+        """
+        Check that the cached dependencies are used instead of downloading them from repo again.
+
+        Preconditions:
+        * On git instance prepare an empty repository
+
+        Process:
+        * Clone the package from the upstream repository
+        * Create empty commit on new test branch and push it to the prepared repository
+        * Send new request to Cachito API which would fetch data from the prepared repository
+        * Delete branch with the corresponding commit
+        * Send the same request to Cachito API
+
+        Checks:
+        * Check that the state of the first request is complete
+        * Check that the commit is not available in the repository after the branch is deleted
+        * Check that the state of the second request is complete
+        """
+        if test_env["cached_dependencies"]["test_repo"].get("use_local"):
+            # create bare git repo
+            bare_repo_dir = Path(test_env["cached_dependencies"]["test_repo"]["ssh_url"])
+            bare_repo = git.Repo.init(str(bare_repo_dir), bare=True)
+            assert bare_repo.bare
+            # We need to expand this for later usage from the original repo directory
+            test_env["cached_dependencies"]["test_repo"]["ssh_url"] = str(bare_repo_dir.resolve())
```

Comment:
If we're replacing the `ssh_url` anyways, could we just do this in the `setup_method`?

Reply:
We would need to create additional fixtures to use `test_env` in the setup method (and use a custom setup method). That is why I ended up not manipulating test_env in the setup. Do you think it is worth to have this additional fixture here?

Reply:
Oh, I see what you mean. I think a new fixture would make it cleaner, yes. This could be defined in this file so it's not accidentally used by other tests.

Reply:
Done. I also extracted the function to create the bare git repository.